lib/usd/ui/layerEditor/layerTreeView.cpp

```diff
@@ -136,6 +138,61 @@ void LayerTreeView::onItemDoubleClicked(const QModelIndex& index)
     }
 }
 
+bool LayerTreeView::shouldExpandOrCollapseAll() const
+{
+    return ((QGuiApplication::keyboardModifiers() & Qt::ShiftModifier) != 0);
```

Comment:
I'm not sure we should be calling Qt directly here to get the keyboard modifiers. Instead I would suggest using the maya command `getModifiers`

Reply:
Is it a C++ command? going through MEL would seem inefficient. I guess this is a mere mouse click, so efficiency is not really a concern.

Reply:
Do we still plan to share the Layer Editor with 3dsMax? Going through a Maya command would make porting harder.

Reply:
I think the keyboard modifiers are passed on in a mouse click event (ex Line 448 ). Perhaps it can be captured then ?

Reply:
I didn't think about the porting, but the LayerEditor already uses other stuff from MGlobal, so we'll deal with it all later.